packages/@ember/-internals/glimmer/lib/utils/args-proxy.ts

```diff
@@ -101,7 +101,7 @@ if (HAS_NATIVE_PROXY) {
 
         const parsed = convertToInt(prop);
 
-        if (parsed !== null && parsed < positional.length) {
+        if (parsed !== null && parsed >= 0 && parsed < positional.length) {
```

Comment:
I don't think this fix is quite enough (we have the exact same type of check in `has`).